### PR TITLE
[Sprint 4] AutoDelete Old Posts Background Job

### PR DESCRIPTION
**Pull request by maxbalves**
## Description
- Every day at 7:00am UTC, the CloudJob `removeOldPosts` in the Parse Server will be automatically ran
    - Posts last updated more than 2 days ago will be deleted, including their comments
    - The lifetime of a post is very small in order to allow the developer (me) to verify posts are being deleted on a daily basis and correctly
- The code sample in the commit is not the file that is ran by the app. It's only to show what Parse Server will run

## Milestones
Milestone/Sprint 4, [#38 Autodelete posts too old](https://github.com/maxbalves/PostBy/issues/38)

## Test Plan
- Created multiple posts with comments and likes from two different users
- Started the job to run every minute
- Kept creating more posts
- Watched as posts older than 1 minute were deleted
- More testing can be done passively as posts will be deleted daily

## Future Plans
- My next goal is to look into Parse's LiveQuery and what features of the app can potentially be changed to use it. In addition, I will need to weight the pro's and con's of the change and consider the time left for development
- Other possible goals:
    - Implement screen for users to see their likes, dislikes, comments, and posts
    - Increase the lifetime of posts instead of only 2 days since update
    - Design app UI

### CODE REVIEW

File: PostBy/CloudCodeSample.js [hunk 1]
```diff
@@ -79,3 +79,38 @@ Parse.Cloud.define("deleteAccount", async (request) => {
   // delete account
   await user.destroy({useMasterKey : true});
 });
+
+// Delete Old Posts - Cron Job
+Parse.Cloud.job("removeOldPosts", async (request) => {
+  let date = new Date();
+  let timeNow = date.getTime();
+  // Posts updatedAt older than: 2 days
+  // 2 (days) * 24 (hours) * 60 (minutes) * 60 (seconds) * 1000 (milliseconds)
```
Comment (raozhang-fb) [inline PostBy/CloudCodeSample.js:88]: Great documentation! 👍

File: PostBy/CloudCodeSample.js [hunk 2]
```diff
@@ -79,3 +79,38 @@ Parse.Cloud.define("deleteAccount", async (request) => {
   // delete account
   await user.destroy({useMasterKey : true});
 });
+
+// Delete Old Posts - Cron Job
+Parse.Cloud.job("removeOldPosts", async (request) => {
+  let date = new Date();
```
Comment (raozhang-fb) [inline PostBy/CloudCodeSample.js:85]: Is this a constant variable? as well as variables below this line?

Usually we don't expect date/timestamps extracted from system can be altered.

File: PostBy/CloudCodeSample.js [hunk 3]
```diff
@@ -79,3 +79,38 @@ Parse.Cloud.define("deleteAccount", async (request) => {
...
+  let Post = Parse.Object.extend("Post");
+  let query = new Parse.Query(Post);
+  
+  // Query posts last modified more than 1 minute ago
+  query.lessThanOrEqualTo("updatedAt", queryDate);
+  
+  const results = await query.find({useMasterKey : true});
+  
+  // Loop through and delete posts + their comments
+  for (const post of results) {
```
Comment (raozhang-fb) [inline PostBy/CloudCodeSample.js:106]: NIT: consider this potential performance improvement -- if we don't care the order of posts being deleted, is it possible to do it in parallel, like asynchronously?

File: PostBy/Global Variables/GlobalVars.m [hunk 4]
```diff
@@ -32,3 +32,5 @@
 NSString *const LIKES_RELATION = @"likes";
 NSString *const DISLIKES_RELATION = @"dislikes";
 
+// !IMPORTANT: Any changes made to fields here will need to be manually done in Parse Server's Background Jobs
```
Comment (raozhang-fb) [inline PostBy/Global Variables/GlobalVars.m:35]: This warning is important but with ambiguity. 

Does "here" mean the fields above this line or below this line? If this file has thousands of lines and a developer only updates the first few lines, does the developer have the chance to notice this warning?

**Review by raozhang-fb — COMMENTED** [review]
Overall it looks good to me!

File: PostBy/Global Variables/GlobalVars.m (hunk 4 above)
Comment (maxbalves) [reply PostBy/Global Variables/GlobalVars.m:35]: Good point Rao! I will put the comment at the top of the file and specific "fields below"

File: PostBy/CloudCodeSample.js (hunk 3 above)
Comment (maxbalves) [reply PostBy/CloudCodeSample.js:106]: Hi Rao! `.destroy()` returns a Parse.Promise, therefore it's asynchronous already. The loop can still go to the next post without needing to wait for the `.destroy()` function to complete

File: PostBy/CloudCodeSample.js (hunk 2 above)
Comment (maxbalves) [reply PostBy/CloudCodeSample.js:85]: It is! I will make sure to change them appropriately.

**Pull request merged**